metrics/lib/features/dashboard/presentation/state/project_metrics_store.dart

```diff
@@ -1,74 +1,140 @@
 
-  int get averageBuildTime => _buildMetrics?.averageBuildTime?.inMinutes;
+        return remove;
+      });
 
-  int get totalBuildNumber => _buildMetrics?.totalBuildNumber;
+      for (final project in projects) {
+        final projectId = project.id;
 
-  /// Load the coverage metric.
-  Future<void> getCoverage(String projectId) async {
-    _coverage = await _getCoverage(ProjectIdParam(projectId: projectId));
+        ProjectMetrics projectMetrics =
+            projectsMetrics[projectId] ?? ProjectMetrics();
+
+        // update name in case it was updated on server
+        projectMetrics = projectMetrics.copyWith(
+          projectName: project.name,
+        );
+
+        if (!projectsMetrics.containsKey(projectId)) {
+          _subscribeToBuildMetrics(projectId);
+        }
+        projectsMetrics[projectId] = projectMetrics;
+      }
+
+      _projectsMetricsSubject.add(projectsMetrics);
+    });
   }
 
-  /// Loads the build metrics.
-  Future<void> getBuildMetrics(String projectId) async {
-    _buildMetrics = await _getBuildMetrics(
-      ProjectIdParam(projectId: projectId),
+  /// Subscribes to project metrics.
+  void _subscribeToBuildMetrics(String projectId) {
+    final buildMetricsStream = _receiveBuildMetricsUpdates(
+      BuildMetricsLoadingParam(
+        projectId,
+        defaultBuildMetricsLoadingPeriod,
+        defaultNumberOfBuildResults,
+      ),
     );
 
-    if (_buildMetrics == null) return;
+    // We are storing subscriptions to map to cancel them later,
+    // but the analyzer can't handle this, so we should add this ignoring.
+    // ignore: cancel_subscriptions
+    final metricsSubscription = buildMetricsStream.listen((metrics) {
+      _createBuildMetrics(metrics, projectId);
+    });
+
+    _buildMetricsSubscriptions[projectId] = metricsSubscription;
+  }
+
+  /// Create project metrics form build metrics.
+  void _createBuildMetrics(BuildMetrics buildMetrics, String projectId) {
+    final projectsMetrics = _projectsMetricsSubject.value;
+
+    final projectMetrics = projectsMetrics[projectId];
+
+    if (projectMetrics == null || buildMetrics == null) return;
 
-    _getPerformanceMetrics();
-    _getBuildNumberMetrics();
-    _getBuildResultMetrics();
+    final performanceMetrics = _getPerformanceMetrics(buildMetrics);
+    final buildNumberMetrics = _getBuildNumberMetrics(buildMetrics);
+    final buildResultMetrics = _getBuildResultMetrics(buildMetrics);
+
+    projectsMetrics[projectId] = projectMetrics.copyWith(
+      performanceMetrics: performanceMetrics,
+      buildNumberMetrics: buildNumberMetrics,
+      buildResultMetrics: buildResultMetrics,
+      totalBuildsNumber: buildMetrics.totalBuildsNumber,
+      averageBuildTime: buildMetrics.averageBuildTime.inMinutes,
+      coverage: buildMetrics.coverage,
+      stability: buildMetrics.stability,
+    );
+
+    _projectsMetricsSubject.add(projectsMetrics);
   }
 
```

Comment:
Should we update the documentation for the functions below according to updates?